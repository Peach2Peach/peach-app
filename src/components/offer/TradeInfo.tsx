import { View, ViewStyle } from 'react-native'
import tw from '../../styles/tailwind'
import { Text } from '../text'

type Props = ComponentProps & {
  text: string
  textStyle?: ViewStyle | ViewStyle[]
  IconComponent?: JSX.Element
}
export const TradeInfo = ({ text, textStyle, IconComponent, style }: Props) => (
  <View style={[tw`flex-row items-center gap-1`, style]}>
    <Text style={[tw`uppercase button-medium`, textStyle]}>{text}</Text>
    {IconComponent}
  </View>
)
