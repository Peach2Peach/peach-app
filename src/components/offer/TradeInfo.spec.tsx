import { createRenderer } from 'react-test-renderer/shallow'
import { TradeInfo } from './TradeInfo'

describe('TradeInfo', () => {
  const renderer = createRenderer()

  it('renders correctly', () => {
    renderer.render(<TradeInfo text={'text'} iconId="alertCircle" />)
    expect(renderer.getRenderOutput()).toMatchSnapshot()
  })
  it('renders correctly with icon color', () => {
    renderer.render(<TradeInfo text={'text'} iconId="alertCircle" iconColor={'#BADA55'} />)
    expect(renderer.getRenderOutput()).toMatchSnapshot()
  })
})
