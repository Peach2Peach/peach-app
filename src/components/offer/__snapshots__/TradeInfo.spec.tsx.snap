// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TradeInfo renders correctly 1`] = `
<View
  style={
    [
      {
        "alignItems": "center",
        "flexDirection": "row",
        "gap": 4,
      },
      undefined,
    ]
  }
>
  <PeachText
    style={
      [
        {
          "fontFamily": "Baloo2-SemiBold",
          "fontSize": 14,
          "letterSpacing": 0.5,
          "lineHeight": 24,
          "textTransform": "uppercase",
        },
        undefined,
      ]
    }
  >
    text
  </PeachText>
</View>
`;

exports[`TradeInfo renders correctly with IconComponent 1`] = `
<View
  style={
    [
      {
        "alignItems": "center",
        "flexDirection": "row",
        "gap": 4,
      },
      {
        "marginTop": 16,
      },
    ]
  }
>
  <PeachText
    style={
      [
        {
          "fontFamily": "Baloo2-SemiBold",
          "fontSize": 14,
          "letterSpacing": 0.5,
          "lineHeight": 24,
          "textTransform": "uppercase",
        },
        undefined,
      ]
    }
  >
    text
  </PeachText>
  <Icon
    id="activity"
  />
</View>
`;

exports[`TradeInfo renders correctly with additional styles 1`] = `
<View
  style={
    [
      {
        "alignItems": "center",
        "flexDirection": "row",
        "gap": 4,
      },
      {
        "marginTop": 16,
      },
    ]
  }
>
  <PeachText
    style={
      [
        {
          "fontFamily": "Baloo2-SemiBold",
          "fontSize": 14,
          "letterSpacing": 0.5,
          "lineHeight": 24,
          "textTransform": "uppercase",
        },
        {
          "color": "#F56522",
        },
      ]
    }
  >
    text
  </PeachText>
</View>
`;
